sdk/communication/azure-communication-chat/src/test/java/com/azure/android/communication/chat/signaling/TrouterUtilsTest.java

```diff
@@ -0,0 +1,132 @@
+
+        assertNotNull(communicationIdentifier);
+        assertTrue(communicationIdentifier instanceof MicrosoftTeamsUserIdentifier);
+        MicrosoftTeamsUserIdentifier microsoftTeamsUserIdentifier =
+            (MicrosoftTeamsUserIdentifier) communicationIdentifier;
+        assertEquals(CommunicationCloudEnvironment.DOD, microsoftTeamsUserIdentifier.getCloudEnvironment());
+        assertFalse(microsoftTeamsUserIdentifier.isAnonymous());
+        assertEquals(USER_ID, microsoftTeamsUserIdentifier.getUserId());
+        assertEquals(teamsDodUserMri, microsoftTeamsUserIdentifier.getRawId());
+    }
+
+    @Test
+    public void canParseTeamsGcchUserMri() {
+        final String teamsGcchUserMri = "8:gcch:" + USER_ID;
+        CommunicationIdentifier communicationIdentifier = TrouterUtils.getCommunicationIdentifier(teamsGcchUserMri);
+
+        assertNotNull(communicationIdentifier);
+        assertTrue(communicationIdentifier instanceof MicrosoftTeamsUserIdentifier);
+        MicrosoftTeamsUserIdentifier microsoftTeamsUserIdentifier =
+            (MicrosoftTeamsUserIdentifier) communicationIdentifier;
+        assertEquals(CommunicationCloudEnvironment.GCCH, microsoftTeamsUserIdentifier.getCloudEnvironment());
+        assertFalse(microsoftTeamsUserIdentifier.isAnonymous());
+        assertEquals(USER_ID, microsoftTeamsUserIdentifier.getUserId());
+        assertEquals(teamsGcchUserMri, microsoftTeamsUserIdentifier.getRawId());
+    }
+
+    @Test
+    public void canParseTeamsVisitorUserMri() {
+        final String teamsVisitorUserMri = "8:teamsvisitor:" + USER_ID;
+        CommunicationIdentifier communicationIdentifier = TrouterUtils.getCommunicationIdentifier(teamsVisitorUserMri);
+
+        assertNotNull(communicationIdentifier);
+        assertTrue(communicationIdentifier instanceof MicrosoftTeamsUserIdentifier);
+        MicrosoftTeamsUserIdentifier microsoftTeamsUserIdentifier =
+            (MicrosoftTeamsUserIdentifier) communicationIdentifier;
+        assertEquals(CommunicationCloudEnvironment.PUBLIC, microsoftTeamsUserIdentifier.getCloudEnvironment());
+        assertTrue(microsoftTeamsUserIdentifier.isAnonymous());
+        assertEquals(USER_ID, microsoftTeamsUserIdentifier.getUserId());
+        assertEquals(teamsVisitorUserMri, microsoftTeamsUserIdentifier.getRawId());
+    }
+
+    @Test
+    public void canParseAcsUserMri() {
+        final String acsUserMri = "8:acs:" + USER_ID;
+        CommunicationIdentifier communicationIdentifier = TrouterUtils.getCommunicationIdentifier(acsUserMri);
+
+        assertNotNull(communicationIdentifier);
+        assertTrue(communicationIdentifier instanceof CommunicationUserIdentifier);
+        CommunicationUserIdentifier communicationUserIdentifier =
+            (CommunicationUserIdentifier) communicationIdentifier;
+        assertEquals(acsUserMri, communicationUserIdentifier.getId());
+    }
+
+    @Test
+    public void canParseSpoolUserMri() {
```

Comment:
Why do we still have 8:spool here? These must be retired, are not they?

Reply:
It should. But every time when I ask Jiri Sofka to check whether we've already deleted all spool resources created during private preview, he said this work is still not done yet. So, I can't say they are all retired...